cmd/ethereum_rust/ethereum_rust.rs

```diff
@@ -111,17 +113,31 @@ async fn main() {
         node_id: local_node_id,
     };
 
+    // TODO: Check every module starts properly.
+    let tracer = TaskTracker::new();
     let rpc_api = ethereum_rust_rpc::start_api(
         http_socket_addr,
         authrpc_socket_addr,
         store,
         jwt_secret,
         local_p2p_node,
-    );
+    )
+    .into_future();
     let networking =
-        ethereum_rust_net::start_network(udp_socket_addr, tcp_socket_addr, bootnodes, signer);
-
-    try_join!(tokio::spawn(rpc_api), tokio::spawn(networking)).unwrap();
+        ethereum_rust_net::start_network(udp_socket_addr, tcp_socket_addr, bootnodes, signer)
+            .into_future();
+
+    tracer.spawn(rpc_api);
+    tracer.spawn(networking);
+
+    tokio::select! {
+        _ = tokio::signal::ctrl_c() => {
+            info!("Server shut down started...");
+            tokio::time::sleep(Duration::from_secs(10)).await;
```

Comment:
10 seconds seems to much, I think 1 should be enough

Reply:
Done!